pdns/recursordist/rec-taskqueue.cc

```diff
@@ -21,18 +21,62 @@
  */
 #include "rec-taskqueue.hh"
 #include "taskqueue.hh"
+#include "logger.hh"
+#include "stat_t.hh"
 #include "syncres.hh"
 
 static thread_local pdns::TaskQueue t_taskQueue;
+static pdns::stat_t s_almost_expired_tasks_pushed;
+static pdns::stat_t s_almost_expired_tasks_run;
+static pdns::stat_t s_almost_expired_tasks_exceptions;
+
+static void resolve(const struct timeval& now, bool logErrors, const pdns::ResolveTask& task)
+{
+  ++s_almost_expired_tasks_run;
```

Comment:
I wonder if we should only increase that counter after the task was successfully completed? That way we would have `almost-expired-pushed` - `currently queued tasks` == `almost-expired-run` + `almost-expired-exceptions`. It does not matter much, to be fair.